lib/covDev.cu

```diff
@@ -37,7 +37,7 @@ namespace quda
 
       constexpr bool xpay = false;
       constexpr int nParity = 2;
-      Dslash::template instantiate<packShmem, nParity, xpay>(tp, stream);
+      Dslash::template instantiate<packStaggeredShmem, nParity, xpay>(tp, stream);
```

Comment:
@maddyscientist  I think you are right. While the code has been refocused towards staggered fermions, it still retains components that are implemented for nSpin = 4, e.g. 
https://github.com/lattice/quda/blob/feature/staggered_correlators/lib/covDev.cu#L138
etc. 
We could put if .. else ... block for each case , to take into account both versions. Any thoughts?

Reply:
Yes, I think making if `if constexpr (Arg::nSpin) ....else...` would be the way to go.

Reply:
Actually, I was wrong regarding my previous comment since this operator is spin -agnostic (no spin related packing or projection). So I think we just keep it as is

Reply:
Agree, I think this should be fine.  Marking as resolved.